e2e/integration_tests/data_flow_test.go

```diff
@@ -464,13 +519,40 @@ func Test_QuerierAvailableToPrometheus(t *testing.T) {
 
 func Test_MetricsWrittenToPostgresDB(t *testing.T) {
 	var err error
-	response_time_row := store.DB.QueryRow("SELECT * FROM metrics_labels WHERE metric_name = 'AllEndpoints_http_response_time';")
-	var id, metric_name, result_label string
-	err = response_time_row.Scan(&id, &metric_name, &result_label)
+	ctx := context.Background()
+	response_time_rows, err := store.DB.QueryContext(ctx, "SELECT * FROM metrics_labels WHERE metric_name = 'AllEndpoints_http_response_time';")
 	failOnError(err)
 
-	if result_label != "{\"job\": \"FHIRQUERY\", \"orgName\": \"LanternTestOrg\", \"instance\": \"endpoint_querier:3333\"}" {
-		t.Fatalf("LanternTestOrg not found in AllEndpoints_http_response_time metric")
+	// Random set of URLs in the TestEndpointSources, unlikely that all 5 of them will have failed
+	// during a run
+	possibleUrls := [5]string{
+		"https://interconnect.lcmchealth.org/FHIR/api/FHIR/DSTU2/metadata",
+		"https://lmcrcs.lexmed.com/FHIR/api/FHIR/DSTU2/metadata",
+		"https://fhir.healow.com/FHIRServer/fhir/IGCGAD/metadata",
+		"https://eprescribe.mercy.net/PRDFHIRSTL/rvh/api/FHIR/DSTU2/metadata",
+		"https://webproxy.comhs.org/FHIR/api/FHIR/DSTU2/metadata",
+	}
+
+	isInDB := false
+	defer response_time_rows.Close()
+	for response_time_rows.Next() {
+		var id, metric_name, result_label string
+		err = response_time_rows.Scan(&id, &metric_name, &result_label)
+		for _, url := range possibleUrls {
+			if strings.Contains(result_label, url) {
+				expectedResultLabel := fmt.Sprintf("{\"job\": \"FHIRQUERY\", \"url\": \"%s\", \"instance\": \"endpoint_querier:3333\"}", url)
+				if result_label == expectedResultLabel {
+					isInDB = true
+					break
+				}
+			}
+		}
+		if isInDB {
+			break
+		}
+	}
+	if !isInDB {
+		t.Fatalf("https://webproxy.comhs.org/FHIR/api/FHIR/DSTU2/metadata not found in AllEndpoints_http_response_time metric")
```

Comment:
should thhis reference a `url` rather than a specific url?

Reply:
yep, missed it when I was rewriting this!